src/Klipfolio.php

```diff
@@ -69,19 +69,31 @@ protected function create(BaseApiResourceConnector $connector)
      */
     protected function update(BaseApiResourceConnector $connector)
     {
+        $this->isValidUpdate($connector);
+        return $this->client->sendRequest(
+            'PUT',
+            $connector->getEndpoint(),
+            ['body' => $connector->getDataForUpdate()]
+        );
+    }
+
+
+    /**
+     * @param BaseApiResourceConnector $connector
+     * @return bool
+     * @throws KlipfolioApiException
+     */
+    protected function isValidUpdate(BaseApiResourceConnector $connector)
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The Cyclomatic Complexity of this function "isValidUpdate" is 5 which is greater than 4 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=php%3AS1541)